src/NUnitFramework/framework/Constraints/MsgUtils.cs

```diff
@@ -200,6 +205,42 @@ private static string FormatArray(Array array)
             return sb.ToString();
         }
 
+        private static string TryFormatKeyValuePair(object value)
+        {
+            if (value == null)
+                return null;
+
+            Type valueType = value.GetType();
+            if (!valueType.GetTypeInfo().IsGenericType)
+                return null;
+
+            Type baseValueType = valueType.GetGenericTypeDefinition();
+            if (baseValueType != typeof(KeyValuePair<,>))
+                return null;
+
+            object k = valueType.GetProperty("Key").GetValue(value, null);
+            object v = valueType.GetProperty("Value").GetValue(value, null);
+
+            return FormatKeyValuePair(k, v);
+        }
+
+        private static string FormatKeyValuePair(object key, object value)
+        {
+            StringBuilder s = new StringBuilder();
+            s.Append('[');
+            if (key != null)
+            {
+                s.Append(FormatValue(key));
+            }
```

Comment:
I don't think we want the `if`. `FormatValue(null)` returns `MsgUtils.Fmt_Null` which is `"null"`.